executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -288,13 +288,22 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> WitgenInference<'a, T, F
         row_offset: i32,
     ) -> ProcessResult<T, Variable> {
         // We need to know the selector.
-        if self
+        let Some(selector) = self
             .evaluate(selector, row_offset)
-            .and_then(|s| s.try_to_known().map(|k| k.is_known_one()))
-            != Some(true)
-        {
+            .and_then(|s| s.try_to_known().map(|k| k.try_to_number()))
+            .flatten()
```

Comment:
```suggestion
            .and_then(|s| s.try_to_known())
            .and_then(|k| k.try_to_number()))
```
Clearer IMO.

Reply:
Got a compiler error about lifetimes and capturing temporaries. can try again.